src/Microsoft.Health.Fhir.Core/Features/Search/Parameters/ISearchParameterSupportResolver.cs

```diff
@@ -12,7 +12,10 @@ public interface ISearchParameterSupportResolver
         /// <summary>
         /// Determines if the given search parameter is able to be indexed
         /// </summary>
-        /// <param name="parameterInfo">Search Parameter info</param>
-        (bool Supported, bool IsPartiallySupported) IsSearchParameterSupported(SearchParameterInfo parameterInfo);
+        /// <param name="parameterInfo">The search parameter to check.</param>
+        /// <returns name="Supported">True if one or more matching converters is found.</returns>
+        /// <returns name="IsPartiallySupported">False if all matching converters are found or no matching converters are found, true otherwise.</returns>
+        /// <returns name="ErrorMessage">Provides more detail if a search parameter is not supported.</returns>
+        (bool Supported, bool IsPartiallySupported, string ErrorMessage) IsSearchParameterSupported(SearchParameterInfo parameterInfo);
```

Comment:
Three return values for a method seem like a lot.  Not sure if we need to change it, but perhaps these should be contained in a struct? or other method.  Just a suggestion.